nbs/15_gsa_gls/07-compile_gwas_snps_and_twas_genes.ipynb

```diff
@@ -0,0 +1,2355 @@
+   "cell_type": "markdown",
+   "id": "bed7a46c-4906-40b9-9dd8-0bfb584e6d2a",
+   "metadata": {
+    "papermill": {
+     "duration": 0.016156,
+     "end_time": "2021-04-28T22:27:17.704727",
+     "exception": false,
+     "start_time": "2021-04-28T22:27:17.688571",
+     "status": "completed"
+    },
+    "tags": []
+   },
+   "source": [
+    "This extend the previous file with more columns"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "850c554f-2ace-40a3-a33a-477246bb07c7",
+   "metadata": {},
+   "source": [
+    "### Get gene's objects"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "dfd93711-5c91-4cb6-8561-f366e270b6b1",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "spredixcan_gene_obj = {\n",
+    "    gene_id: Gene(ensembl_id=gene_id) for gene_id in spredixcan_genes_models.index\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "59efbedc-0b5d-4d0b-bd82-95e74c0ba793",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "len(spredixcan_gene_obj)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "27ac1828-755a-424a-b416-6aef1f0c2e45",
+   "metadata": {},
+   "source": [
+    "### Add genes' variance captured by principal components"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "893b3418-054c-44d2-9ce0-7fa590aef00e",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def _get_gene_pc_variance(gene_row):\n",
+    "    gene_id = gene_row.name\n",
+    "    gene_tissues = gene_row[\"tissue\"]\n",
+    "    gene_obj = spredixcan_gene_obj[gene_id]\n",
+    "\n",
+    "    u, s, vt = gene_obj.get_tissues_correlations_svd(\n",
+    "        tissues=gene_tissues,\n",
+    "        snps_subset=gwas_variants_ids_set,\n",
+    "        reference_panel=REFERENCE_PANEL,\n",
+    "        model_type=EQTL_MODEL,\n",
+    "        # use_covariance_matrix=True,\n",
+    "    )\n",
+    "\n",
+    "    return s"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "083afa33-cea0-40d3-af20-94a1307a6607",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "_tmp = spredixcan_genes_models.loc[\"ENSG00000188976\"]\n",
+    "_get_gene_pc_variance(_tmp)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "80f7720d-79c9-4183-bb7e-b1480f31ed87",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "spredixcan_genes_tissues_pc_variance = spredixcan_genes_models.apply(\n",
+    "    _get_gene_pc_variance, axis=1\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "621302a1-1407-4400-9282-e37798bb18b3",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "spredixcan_genes_tissues_pc_variance"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "9ac9e0fb-e0cc-4fea-ad01-2e76d40afb3c",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# # testing\n",
```

Comment:
Does "testing" mean this is for debugging and should go away? If not maybe make it clearer what it's testing?

Reply:
Yes, updated!